src/withForm.ts

```diff
@@ -133,49 +135,31 @@ const withFormExtended = <T extends Constructor<ConnectableElement & FormConfig>
       }
     }
 
-    // TODO: Check remove recursion. I think === 'string' was missing a return.
     // Set touched for name and handleValidate if needed
-    handleBlur = (nameOrEvent: string | Event): void => {
-      let name: string;
-
-      // Extracts the name from the event.
-      if (typeof nameOrEvent !== 'string') {        
-        const eventTarget = getEventTarget(nameOrEvent)
+    handleBlur = (event: Event | CustomEvent, name?: string): void => {
```

Comment:
For `handleBlur` and `handleChange` I have added a second parameter to be able to override the name when called from `withField`. It allow for some code simplification while keeping types simple.

Reply:
I understand that in JS it's "common" to have a function that has different behavior according to the parameters... because we can.

However I'm not sure if it is the best design.

For example the former `handleChange` would return a setter when called with a name and void if called with an event.

Maybe:

```
setTouched(name: string): void {
  this._touched = set(this._touched, actualName, true)
  if (this._validateOnBlur) {
    this.handleValidate()
  }    
}

handleBlur(event) {
  const eventTarget = getEventTarget(event)
  if (eventTarget) {
    this.setTouched(eventTarget.name);
  }
}
```

This way:
- `handle<EventName>()` does what it says it does: handling the event,
- you can still call `setTouched(...)` if you do not have an event (cf your comments in `withField`)
- would be same way for `handleChange`, extracting the logic to `setValueAndValidate(name, value)`

Reply:
Alternate proposal:

1. I think that for consistency `withField.handleBlur()` should call `withField.setTouched(name || id)` that will delegate to `this._formClass.setTouched(...)`

2. May be we `setValue` can be be changed to accept a validation param `.setValue(value, validation = false)`. Also to be consistent `setValue(value, validation)` would be implemented on both `withField` and `withForm`. The former would delegate to the later. 

3. `withField.handleChange(event)` would call `withField.setValue(value, true)` which would delegate to `withForm.setValue(name || id, value, true)`

4. `handle<EventName>(event)` methods are listeners that only take args.

The difference in my proposal is that `setTouched` and `setValue` are also implemented in `withField`. That is mostly for consistency so that `withField` and `withForm` have a similar. 
For now I am only using `withForm` in my code so you have much more experience to decide which of the proposal makes more sense. I'll implement whatever you prefer.

Thanks

Reply:
Great idea! Really like it all!

Reply:
Thanks !

I'll work on the changes described here this week.
Once everything looks good, I'll polish the CL (adding cjs, module and umd outputs, updating the doc, ...).

Reply:
Done